packages/scalecube-microservice/src/ServiceCall/RemoteCall.ts

```diff
@@ -26,7 +28,71 @@ export const remoteCall = ({
     }) as Observable<any>;
   }
 
-  // TODO remote invoke
-  // TODO if service is remote then use transport else invoke the function
-  return of({});
+  return remoteResponse({ address: endPoint.address, asyncModel, message, openConnections });
+};
+
+const remoteResponse = ({
+  address,
+  asyncModel,
+  message,
+  openConnections,
+}: {
+  address: string;
+  asyncModel: string;
+  message: Message;
+  openConnections: { [key: string]: any };
+}) => {
+  let connection: any;
+  if (!openConnections[address]) {
+    const client = CreateClient({ address });
+    connection = client.connect();
+    openConnections[address] = connection;
+  } else {
+    connection = openConnections[address];
+  }
+
+  return new Observable((observer) => {
+    try {
+      connection.then((socket: any) => {
+        const serializeData = JSON.stringify(message);
+        const socketConnect = socket[asyncModel]({
+          data: serializeData,
+          metadata: '',
+        });
+
+        switch (asyncModel) {
+          case ASYNC_MODEL_TYPES.REQUEST_RESPONSE:
+            socketConnect.then((response: RsocketEventsPayload) => {
+              const { data } = response && JSON.parse(response.data);
```

Comment:
the callback for both async models is the same - I suggest to extract it to a common helper